### PR TITLE
docs: explicitly mention deviations from semver

### LINKED ISSUE #1675: v1.4.0-rc.0 contains breaking changes
### What's wrong?

A release candidate for v1.4.0 was cut, but it contains breaking changes.

As alloy uses semver, these breaking changes must lead to a major version increase, so the tag should be for `v2.0.0-rc.0` instead, leading up to a release for `v2.0.0`.

### Steps to reproduce

Check out the [breaking changes](https://github.com/grafana/alloy/compare/v1.3.1...v1.4.0-rc.0#diff-06572a96a58dc510037d5efa622f9bec8519bc1beab13c9f251e97e657a9d4edR12) section.

### System information

_No response_

### Software version

_No response_

### Configuration

_No response_

### Logs

_No response_

### PR DESCRIPTION
**Pull request by morremeyer**
#### PR Description

This explicitly documents that the versioning scheme is not exactly following semantic versioning, but trying to keep as close as possible to it.

#### Which issue(s) this PR fixes

Fixes #1675

#### PR Checklist

- [x] Documentation added

### CODE REVIEW

File: docs/sources/introduction/backward-compatibility.md [hunk 1]
```diff
@@ -8,7 +8,7 @@ weight: 999
 
 # {{% param "FULL_PRODUCT_NAME" %}} backward compatibility
 
-{{< param "FULL_PRODUCT_NAME" >}} follows [semantic versioning][].
+{{< param "FULL_PRODUCT_NAME" >}} mosty follows [semantic versioning][], with the differences documented in the [Exceptions](#exceptions) section.
```
Comment (wildum) [inline docs/sources/introduction/backward-compatibility.md:11]:
```suggestion
{{< param "FULL_PRODUCT_NAME" >}} mostly follows [semantic versioning][], with the differences documented in the [Exceptions](#exceptions) section.
```

**Review by wildum — APPROVED** [review]
This is fine by me, waiting for @clayton-cornell input

File: docs/sources/introduction/backward-compatibility.md [hunk 2]
```diff
@@ -8,7 +8,7 @@ weight: 999
 
 # {{% param "FULL_PRODUCT_NAME" %}} backward compatibility
 
-{{< param "FULL_PRODUCT_NAME" >}} follows [semantic versioning][].
+{{< param "FULL_PRODUCT_NAME" >}} mostly follows [semantic versioning][], with the differences documented in the [Exceptions](#exceptions) section.
```
Comment (clayton-cornell) [inline docs/sources/introduction/backward-compatibility.md:11]:
```suggestion
{{< param "FULL_PRODUCT_NAME" >}} follows [semantic versioning][]. [Exceptions](#exceptions) may arise that require a breaking change without a new major version.
```

I think we should avoid the vague "mostly" and just state we use semver, but exceptions can happen.

File: docs/sources/introduction/backward-compatibility.md [hunk 3]
```diff
@@ -21,7 +21,7 @@ Documented functionality that's released as _Generally available_ is covered by
 
 ## Exceptions
 
-We strive to maintain backward compatibility, but there are situations that may arise that require a breaking change without a new major version:
+We strive to maintain backward compatibility, but there are situations that may arise that require a breaking change without a new major version, deviating from [item 8 of the semver specification]:
```
Comment (clayton-cornell) [inline docs/sources/introduction/backward-compatibility.md:24]:
```suggestion
We strive to maintain backward compatibility, but there are situations that may arise that require a breaking change without a new major version, deviating from [item 8 of the semver specification][]:
```

**Review by clayton-cornell — APPROVED** [review]
One suggestion, otherwise it's fine :-)

**Review by wildum — APPROVED** [review]

**Comment by clayton-cornell** [comment]
Thanks for this change @morremeyer   Getting this change merged and backported today.

**Pull request merged**